File: Python/10-third-greatest.py
def third_greatest(nums):
    first = 0
    second = 0
    third = 0

    for i in range(0, len(nums)):
        if nums[i] > first:
            third = second
            second = first
            first = nums[i]
        elif nums[i] > second:
            third = second
            second = nums[i]
        elif nums[i] > third:
            third = nums[i]

    return third

print(third_greatest([3,4,5,8,2]))

File: Javascript/10-third-greatest.js
function thirdGreatest(nums) {
	var first = 0, second = 0, third = 0;

	for (var i = 0; i < nums.length; i++) {
		if (nums[i] > first || first == 0) {
			third = second;
			second = first;
			first = nums[i];
		} else if (nums[i] > second || second == 0) {
			third = second;
			second = nums[i];
		} else if (nums[i > third || third == 0]) {
			third = nums[i];
		}
	}

	return third;
}

console.log(thirdGreatest([5,3,7,4,5,6,0,0,0,0,0]));
